Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -43,71 +57,72 @@
           "Selector": "$.Reservations[0].Instances[0].State.Name",
           "Type": "String"
         }
-      ],
-      "name": "describeInstance",
-      "action": "aws:executeAwsApi",
-      "timeoutSeconds": 30,
-      "onFailure": "Abort"
+      ]
     },
 
     {
-       "name": "describeInstanceRootVolume",
-       "action": "aws:executeAwsApi",
-       "onFailure": "Abort",
-       "timeoutSeconds": 60,
-       "maxAttempts": 10,
-       "inputs": {
-         "Service": "ec2",
-         "Api": "DescribeVolumes",
-         "Filters": [
-           {
-              "Name": "attachment.instance-id",
-              "Values": [
-                "{{instanceId}}"
-              ]
-           },
-           {
-              "Name": "attachment.device",
-              "Values": [
-                "{{describeInstance.rootDeviceName}}"
-              ]
-           }
-         ]
-       },
-       "outputs": [
-         {
-            "Name": "rootDeviceVolumeId",
-            "Selector": "$.Volumes[0].Attachments[0].VolumeId",
-            "Type": "String"
-         },
-         {
-            "Name": "rootDeviceVolumeType",
-            "Selector": "$.Volumes[0].VolumeType",
-            "Type": "String"
-         },
-         {
-            "Name": "RootDeviceDeleteOnTermination",
-            "Selector": "$.Volumes[0].Attachments[0].DeleteOnTermination",
-            "Type": "Boolean"
-         }
-       ],
-       "isCritical": true
+      "name": "describeInstanceRootVolume",
+      "action": "aws:executeAwsApi",
+      "onFailure": "Abort",
+      "timeoutSeconds": 60,
+      "maxAttempts": 10,
+      "isCritical": true,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "DescribeVolumes",
+        "Filters": [
+          {
+            "Name": "attachment.instance-id",
+            "Values": [
+              "{{instanceId}}"
+            ]
+          },
+          {
+            "Name": "attachment.device",
+            "Values": [
+              "{{describeInstance.rootDeviceName}}"
+            ]
+          }
+        ]
+      },
+      "outputs": [
+        {
+          "Name": "rootDeviceVolumeId",
+          "Selector": "$.Volumes[0].Attachments[0].VolumeId",
+          "Type": "String"
+        },
+        {
+          "Name": "rootDeviceVolumeType",
+          "Selector": "$.Volumes[0].VolumeType",
+          "Type": "String"
+        },
+        {
+          "Name": "RootDeviceDeleteOnTermination",
+          "Selector": "$.Volumes[0].Attachments[0].DeleteOnTermination",
+          "Type": "Boolean"
+        }
+      ]
     },
 
     {
+      "name": "createSnapshot",
+      "action": "aws:executeAutomation",
+      "timeoutSeconds": 1800,
```

Comment:
Shouldn't this be {{ maxWaitForCreateSnapshot }} ?

Reply:
I missed your comment on this. It sounds like a bug, but try to change the *maxWaitForCreateSnapshot* parameter type to String (even though it is a number) and see if that works.

If that is the case, you can add a regEx to the String parameter that represents the timeout value such as:

[0-9]{1,5}

While not perfect, it will allow for up to 99999 seconds approx 27 hours, which is massive.